frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/pipeline-details/PipelineVisualizationTask.tsx

```diff
@@ -100,6 +100,22 @@ export const PipelineVisualizationTask: React.FC<PipelineVisualizationTaskProp>
   }
 
   let resources;
+  if (task.taskSpec) {
+    return (
+      <TaskComponent
+        pipelineRunName={pipelineRunName}
+        task={{ data: { spec: task.taskSpec } }}
```

Comment:
Since `task` is the only prop that differs from `taskComponent`'s props, maybe it would be easier to check and conditionally pass it to `taskComponent` itself.